provisioning_client/tests/hsm_client_riot_ut/hsm_client_riot_ut.c

```diff
@@ -124,34 +132,47 @@ static char* umocktypes_stringify_RIOT_ECC_PRIVATE(const RIOT_ECC_PRIVATE* value
 static int umocktypes_are_equal_RIOT_ECC_PRIVATE(RIOT_ECC_PRIVATE* left, RIOT_ECC_PRIVATE* right)
 {
     int result;
-    if (left == right)
+    if ((right == NULL) && (left == NULL))
     {
         result = 1;
     }
-    else if ((left == NULL) || (right == NULL))
+    else if (((right == NULL) && (left != NULL)) || ((left == NULL) && (right != NULL)))
+    {
+        result = 0;
+    }
+    else if ((right->s != left->s) || (right->n != left->n) || (right->p != left->p))
     {
         result = 0;
     }
     else
     {
-        result = memcmp(left->data, right->data, sizeof(left->data));
+        result = 1;
     }
     return result;
 }
 
 static int umocktypes_copy_RIOT_ECC_PUBLIC(RIOT_ECC_PUBLIC* dest, const RIOT_ECC_PUBLIC* src)
 {
-    int result;
+    int result = 1;
```

Comment:
same as above about guessing. this should be declaration only but you can also likely just `return 0`

Reply:
Fixed see above